app/scripts/controllers/edit/deploymentConfig.js

```diff
@@ -0,0 +1,289 @@
+                    okButtonText: "Yes",
+                    okButtonClass: "btn-primary",
+                    cancelButtonText: "No"
+                  };
+                }
+              }
+            });
+
+            modalInstance.result.then(function () {
+              // Move parameters that belong to the origial strategy to the picked one.
+              $scope.strategyData[pickedStrategyParams] = $scope.strategyData[getParamsString($scope.originalStrategy)];
+              $scope.paramsMoved = getParamsString($scope.originalStrategy);
+            }, function() {
+              // Create empty parameters for the newly picked strategy
+              $scope.strategyData[pickedStrategyParams] = {};
+            });
+          } else {
+          }
+          break;
+        default:
+          if (!_.has($scope.strategyData, pickedStrategyParams)) {
+            if (pickedStrategy !== 'Custom') {
+              $scope.strategyData[pickedStrategyParams] = {};
+            } else {
+              $scope.strategyData[pickedStrategyParams] = {
+                image: "",
+                command: [],
+                environment: []
+              }
+            } 
+          }
+          break;
+      }
+      $scope.displayedParams = getParamsString(pickedStrategy);
+    };
+
+    var getParamsString = function(strategyType) {
+      return strategyType.toLowerCase() + 'Params';
+    }
+
+    // helper for detemining if strategy switch was done between Rolling <-> Recreate strategy
+    var isRollingRecreateSwitch = function(pickedStrategy) {
+      return (pickedStrategy !== 'Custom' && $scope.originalStrategy !== 'Custom' && pickedStrategy !== $scope.originalStrategy);
+    };
+
+    var assembleImageChangeTrigger = function(containerName, ist, trigger) {
+      var istagObject = {
+        kind: "ImageStreamTag",
+        namespace: ist.namespace,
+        name: ist.imageStream + ':' + ist.tagObject.tag
+      }
+      if (trigger) {
+        trigger.imageChangeParams.from = istagObject;
+      } else {
+        trigger = {
+          type: "ImageChange",
+          imageChangeParams: {
+            automatic: true,
+            containerNames: [containerName],
+            from: istagObject
+          }
+        };
+      }
+      return trigger;
+    };
+
+    var updateTriggers = function() {
+      var updatedTriggers = _.remove($scope.updatedDeploymentConfig.spec.triggers, function(trigger) {return trigger.type !== "ImageChange"});
+      _.each($scope.containersDataMap, function(containerData, containerName) {
+        if (containerData.hasDeploymentTrigger) {
+          updatedTriggers.push(assembleImageChangeTrigger(containerName, containerData.triggerData.istag, containerData.triggerData.data));
+        } else {
+          _.find($scope.updatedDeploymentConfig.spec.template.spec.containers, function(container) {return container.name === containerName}).image = containerData.image;
```

Comment:
Ah I see. I'd split this into two lines with a local var. Too easy to miss that you're setting the image value on the result.